components/builder-graph/src/data_store.rs

```diff
@@ -55,8 +227,41 @@ impl DataStore {
     /// This includes all the schema and data migrations, along with stored procedures for data
     /// access.
     pub fn setup(&self) -> Result<()> { Ok(()) }
+}
+
+impl Unbuildable for DataStore {
+    fn filter_unbuildable(&self,
+                          packages: &[PackageIdentIntern],
+                          target: PackageTarget)
+                          -> Result<Vec<PackageIdentIntern>> {
+        let package_idents: Vec<BuilderPackageIdent> =
+            packages.iter()
+                    .map(|x| BuilderPackageIdent((*x).into()))
+                    .collect();
 
-    pub fn get_job_graph_packages(&self) -> Result<Vec<PackageWithVersionArray>> {
+        let conn = self.pool.get_conn()?;
+
+        let query = origin_projects::table.select(origin_projects::name)
+                                          .filter(origin_projects::auto_build.eq(true))
+                                          .filter(origin_projects::target.eq(target.to_string()))
+                                          .filter(origin_projects::name.eq_any(package_idents));
+
+        // let debug = diesel::query_builder::debug_query::<diesel::pg::Pg, _>(&query);
+
+        let result: Vec<BuilderPackageIdent> = query.get_results(&conn).unwrap();
```

Comment:
Is this unwrap safe, is there any case in which the query selection might return empty?